opennlp-tools/src/test/java/opennlp/tools/util/AbstractDownloadUtilTest.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package opennlp.tools.util;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.junit.jupiter.api.BeforeAll;
+
+import opennlp.tools.EnabledWhenCDNAvailable;
+
+import static org.junit.jupiter.api.Assertions.fail;
+
+@EnabledWhenCDNAvailable(hostname = "dlcdn.apache.org")
+public abstract class AbstractDownloadUtilTest {
+
+  private static final String APACHE_CDN = "dlcdn.apache.org";
+
+  @BeforeAll
+  public static void cleanupWhenOnline() {
+    boolean isOnline;
+    try (Socket socket = new Socket()) {
+      socket.connect(new InetSocketAddress(APACHE_CDN, 80), EnabledWhenCDNAvailable.TIMEOUT_MS);
+      isOnline = true;
+    } catch (IOException e) {
+      // Unreachable, unresolvable or timeout
+      isOnline = false;
+    }
+    // If CDN is available -> go cleanup in preparation of the actual tests
+    if (isOnline) {
+      wipeExistingModelFiles("-tokens-");
```

Comment:
For consistency: Missing "-pos-" and "-lemmas-" here?

Reply:
This test does not use `pos' or `lemma', so it might not be needed at all.. Since the `ci` profile is not enabled by default, it currently deletes OpenNLP models from the devs machine.

In general (regardless of pos / lemma):  It might be a good idea to update these patterns to an `openlp wildcard expression' for models to avoid deleting things of value on dev machines _or_ to change the default for tests, i.e. activate "ci" profile by default.

Reply:
It uses a glob. So all good. Updating.